client/homebrew/editor/stringArrayEditor/stringArrayEditor.jsx

```diff
@@ -109,35 +126,64 @@ const StringArrayEditor = createClass({
 		this.setState({ valueContext, updateValue: '' });
 	},
 
+	handleTagKeyDown : function(event, index) {
+		if(_.includes(['Enter', 'Space'], event.code)){
+			event.preventDefault();
+			this.editValue(index);
+			setTimeout(()=>{
+				event.target.querySelector('input').focus();
+			}, 0);
+		} else if(_.includes(['Delete'], event.key)){
+			this.removeValue(index);
+		} else if(_.includes(['ArrowLeft'], event.key)){
+			event.target.previousElementSibling?.focus();
+		} else if(_.includes(['ArrowRight'], event.key)){
+			event.target.nextElementSibling?.focus();
+		}
+
+
+	},
+
+	renderDatalist : function(){
+		if(this.props.options?.length > 0){
+			return <datalist id={`${this.props.id}__tags-precoordinated`}>
+				{this.props.options.map((option)=>{
+					return <option value={`${option}`}></option>;
+				})}
+			</datalist>;
+		}
+	},
+
 	render : function() {
-		const valueElements = Object.values(this.state.valueContext).map((context, i)=>context.editing
-			? <React.Fragment key={i}>
-				<div className='input-group'>
-					<input type='text' className={`value ${this.valueIsValid(this.state.updateValue, i) ? '' : 'invalid'}`} autoFocus placeholder={this.props.placeholder}
+		const valueElements = Object.values(this.state.valueContext).map((context, i)=>{
+			return <React.Fragment key={i}>
+				<div aria-label={`tag ${context.value}`} className={`tag ${context.editing ? 'editable' : ''}`} tabIndex={-1} onKeyDown={(e)=>this.handleTagKeyDown(e, i)}>
+					<span className={`tag-text ${context.editing ? 'hidden' : 'visible'}`} key={i} onClick={()=>this.editValue(i)}>{context.value}</span>
```

Comment:
If you click on an existing tag to edit it, you can't start typing right away. You have to click again to enter the text box. Is it possible to focus on the textbox with one click?